SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimAnalyzer.cc

```diff
@@ -261,7 +460,522 @@ EcalEBTrigPrimAnalyzer::analyze(const edm::Event& iEvent, const  edm::EventSetup
+	  if ( gpIter->pdgId() == 11 ||  gpIter->pdgId() == -11 ) {
+	    if (abs(gpIter->eta()) < 1.5 && gpIter->pt()>5 ) {
+	      const reco::GenParticle* myGp=&(*gpIter); 
+	      gpEta = etaTransformation(gpIter->eta(), gpIter->vz() );
+
+              double dphi =  phiCl_ - gpIter->phi();
+              if ( std::abs(dphi)>CLHEP::pi) { dphi = dphi<0? (CLHEP::twopi) + dphi : dphi - CLHEP::twopi;}
+              double  deltaR2  = ( etaCl_ - gpEta ) *  ( etaCl_ - gpEta ) + dphi*dphi;
+
+	      if ( deltaR2 < dR ) {
+		dR = deltaR2;
+		gpMatch=myGp;
+	      } 
+	    }
+	  }
+	}
+	if (  gpMatch !=0 && sqrt(dR)<0.1) {
+	  std::cout << "cluTP Et " << etClInGeV_ << " gen ele mom " << gpMatch->p() << " pt " << gpMatch->pt() << " charge " << gpMatch->charge() <<  " eta " << gpMatch->eta() << " phi " << gpMatch->phi() << std::endl;
+	  h_cluTPEtoverGenEt_->Fill ( etClInGeV_/ gpMatch->pt()); 
+          float matchDPhi = phiCl_ - gpMatch->phi();
+	  if ( std::abs(matchDPhi)>CLHEP::pi) { matchDPhi = matchDPhi<0? (CLHEP::twopi) + matchDPhi : matchDPhi - CLHEP::twopi;}
+	  float matchEta = etaTransformation(gpMatch->eta(), gpMatch->vz() );
+          double matchDR = sqrt ( ( etaCl_ - matchEta ) *  ( etaCl_ - matchEta ) + matchDPhi*matchDPhi );
+	  h_dEta_cluTP_gen_ -> Fill( etaCl_ - matchEta);
+	  h_dPhi_cluTP_gen_ -> Fill(matchDPhi);
+	  h_deltaR_cluTPGen_ -> Fill (matchDR);
+
+
+
+	}     
+      }  // if genParticleValid 
+
+      */
+
+      treeCl_->Fill();    
+
+    }
+    std::cout << " nCluTPPerEvt " << nCluTPPerEvt << std::endl;
+    hCluTPperEvt_ ->Fill(float(nCluTPPerEvt));
+
+  }
+
   
+  if ( nClu>0)         h_nClu_[0] ->Fill(float(nClu));
+  if ( nClu_Etgt_05>0) h_nClu_[1] ->Fill(float(nClu_Etgt_05));
+  if ( nClu_Etgt_1>0)  h_nClu_[2] ->Fill(float(nClu_Etgt_1));
+  if ( nClu_Etgt_2>0)  h_nClu_[3] ->Fill(float(nClu_Etgt_2));
+  if ( nClu_Etgt_3>0)  h_nClu_[4] ->Fill(float(nClu_Etgt_3));
+  
+  
+  // match the clusters made of RH with gen
+  float nGenInEB=0;    
+  if ( isGenParticleValid_ )  {
+    reco::GenParticleCollection::const_iterator gpIter ;    
+    reco::GenParticleCollection::const_iterator gpIterMatch ;    
+    reco::Candidate::PolarLorentzVector* cluRH_p4;
+    reco::Candidate::PolarLorentzVector* trueEle_p4=0;
+    reco::Candidate::PolarLorentzVector* matchEle_p4=0;
+    BaseParticlePropagator prop;
+    
+    float dR=9999;
+    nGenInEB=0;    
+    
+    
+    for ( std::vector<PhotonMCTruth>::const_iterator imcEle=mcEle.begin(); imcEle!=mcEle.end(); imcEle++ ) {
+      
+      if (   abs(imcEle->fourMomentum().pseudoRapidity()) >= 1.5 || imcEle->fourMomentum().et() < 10 ) continue;
+      //      if ( imcEle->hasBrem() !=0) continue;
+      
+      if (debug_) std::cout << " imcEle pt " << imcEle->fourMomentum().et() << " E " << imcEle->fourMomentum().e() << std::endl;
+      bool isTheSame=false;
+      for ( gpIter=gp->begin() ; gpIter!=gp->end() ; gpIter++ ) {
+	//	if ( abs(gpIter->pdgId()) != 11 ) continue;
```

Comment:
Delete commented out code, or justify with a comment why you still need to keep it